core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1539,6 +1539,9 @@ abstract class RDD[T: ClassTag](
     // NOTE: we use a global lock here due to complexities downstream with ensuring
     // children RDD partitions point to the correct parent partitions. In the future
     // we should revisit this consideration.
+    if (doCheckpointCalled) {
+      logWarning(s"Because job has been executed on RDD ${id}, checkpoint won't work")
```

Comment:
reping @zsxwing Would you mind take a look? This is a simple PR but it will bring much help for spark developers to avoid them making some mistake usage...